### PR TITLE
Godot Logger as a Godot Plugin.

### LINKED ISSUE #25: Make godot-logger singleton script as a godot addon
Currently, godot-logger is a singleton script that needs to be added manually to the project settings.
I propose to make godot-logger a Godot addon that can be installed from the `AssetLib`.

The plugin can do :
- install logger files to `addons` directory. 
- Automatically once installed, add the script as a singleton to the project.

### PR DESCRIPTION
**Pull request by MrZak-dev**
Closes #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 # Usage
-
-Copy the `logger.gd` file into your project folder, and define it as an autoloaded
-singleton in your project's settings (e.g. with the name *Logger*):
-```
-Logger="*res://logger.gd"
-```
+1. Clone this repository to `godot_project/addons/`.
+2. Enable the addon from  Project -> Project Settings -> Plugins -> Godot Logger.
+3. An autoloaded script will be added to your ptoject singletons list as `Logger`.
```
Comment (akien-mga) [inline README.md]:
```suggestion
3. An autoloaded script will be added to your project singletons list as `Logger`.
```

File: README.md [hunk 2]
```diff
@@ -1,16 +1,13 @@
-Logger - Logging singleton for Godot Engine
+Godot Logger - Logging Addon for Godot Engine
 ===========================================
 
-The *Logger* class is a GDScript singleton that provides a logging API for
+The *Godot Logger* is a plugin that provides a logging API for
 projects developed with [Godot Engine](https://godotengine.org).
 
 # Usage
-
```
Comment (akien-mga) [inline README.md:8]: Please keep the separation line after the heading.

File: README.md [hunk 3]
```diff
@@ -36,7 +33,7 @@ relevant content.
 * `{MSG}`     = Message from the user
 * `{TIME}`    = Timestamp when the logging occurred
 * `{ERR_MSG}` = Error message corresponding to the error code, if provided.
-                It is automatically prepended with a space.
+				It is automatically prepended with a space.
```
Comment (akien-mga) [inline README.md]: Wrong indentation change.

File: plugin.gd [hunk 4]
```diff
@@ -0,0 +1,10 @@
+@tool
+extends EditorPlugin
+
+
+func _enter_tree() -> void:
+	add_autoload_singleton("Logger", "res://addons/godot-logger/logger.gd")
```
Comment (akien-mga) [inline plugin.gd]: Does it work with just `"logger.gd"` as a relative path? In case people clone / copy the repo with a different name.

Comment (MrZak-dev) [reply plugin.gd]: Yeah that works , Thanks !

**Comment by MrZak-dev** [comment]
@akien-mga

File: README.md [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
 projects developed with [Godot Engine](https://godotengine.org).
 
 # Usage
 
-Copy the `logger.gd` file into your project folder, and define it as an autoloaded
-singleton in your project's settings (e.g. with the name *Logger*):
-```
-Logger="*res://logger.gd"
-```
+1. Clone this repository to `godot_project/addons/`.
```
Comment (akien-mga) [inline README.md]: Experience showed me that many users, including developers, don't understand this kind of pseudo path and will take it literally (and then wonder why it doesn't work).

So I would do:
```suggestion
1. Clone or extract this repository as the `res://addons/godot-logger` folder in your project.
```

**Review by akien-mga — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (MrZak-dev) [reply README.md]: Done  👍

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**